### ISSUE TITLE
BeamNG native Linux port randomly crashes whole system only on Hyprland

### ISSUE THREAD

**Issue by nonetrix** [hyprwm/Hyprland#3107]
### Hyprland Version

v0.28.0 to 84c4a14

### Bug or Regression?

Bug

### Description

Let me say something first, BeamNG native Linux port is in early testing and is highly unstable and crashes often under any compositor or XOrg session and even with my old GTX 1080. However, it should not take my system with it ideally. It crashes so badly that even pressing power or switching to TTY sometimes doesn't work. If you don't own this game, I can either do the testing for you if you give me instructions, or I can maybe let you remote into my box

Here are my specs:
```
                   -`                    noah@dingus-pc
                  .o+`                   --------------
                 `ooo/                   OS: Arch Linux x86_64
                `+oooo:                  Host: B450 AORUS PRO WIFI
               `+oooooo:                 Kernel: 6.4.12-arch1-1
               -+oooooo+:                Uptime: 22 mins
             `/:-:++oooo+:               Packages: 874 (pacman)
            `/++++/+++++++:              Shell: bash 5.1.16
           `/++++++++++++++:             Resolution: 1920x1080
          `/+++ooooooooooooo/`           DE: Hyprland
         ./ooosssso++osssssso+`          Theme: adw-gtk3-dark [GTK3]
        .oossssso-````/ossssss+`         Icons: gruvbox-dark-icons-gtk [GTK3]
       -osssssso.      :ssssssso.        Terminal: alacritty
      :osssssss/        osssso+++.       Terminal Font: Jetbrains Mono Nerd Font
     /ossssssss/        +ssssooo/-       CPU: AMD Ryzen 7 3700X (16) @ 3.600GHz
   `/ossssso+/:-        -:/+osssso+-     GPU: AMD ATI Radeon RX 6800/6800 XT / 6900 XT
  `+sso+:-`                 `.-/+oso:    Memory: 3231MiB / 32016MiB
 `++:.                           `-/+/
 .`                                 `/
 ```

### How to reproduce

1. Download BeamNG drive
2. Don't launch via Steam, instead navigate to `SteamLibrary/steamapps/common/BeamNG.drive/BinLinux/`
3. Run the executable and load the game
4. take notice that screen flickers sometimes when loading in fullscreen or just crashes in same way when loading
5. Play the game for a while drive etc.
6. It crashes and brings the whole system with it
7. Usually you will see the last frames, or in some cases it will display a few previous frames and switch between them 

### Crash reports, logs, images, videos

No useful logs generated

Strange artifacts on game boot and resize, this happens on  KDE as well but no crashing the compositor. Likely VRAM related? Also notice that it seems to show some Chinese or kanji fonts weirdly:
![2023-08-29-192509_hyprshot](https://github.com/hyprwm/Hyprland/assets/45698918/1f298cc8-dd17-4c31-b91f-8136df72762e)

Crash itself (note: here I was able to use TTY, but this is not always true)
https://youtu.be/Y6hLHkTqw4I

**Comment by nonetrix** [comment]
Sorry for opening and closing the issue, I couldn't reproduce it for a while. Seems quite random sorry

**Comment by vaxerski** [comment]
does it work on sway-git?

Where is the crash report?

**Comment by nonetrix** [comment]
There is no crash report, but I will test sway-git later

**Comment by UltraBlackLinux** [comment]
https://www.beamng.com/threads/linux-port-%E2%80%93-feedback-known-issues-and-faq.86422/

There is a forum thread for this game. The crashes are just the display driver crashing it seems. It might just be a mesa bug

**Comment by nonetrix** [comment]
Sorry, unexpectedly had to travel out of state. I will test this as soon as I am back :3

**Comment by nonetrix** [comment]
Just reproduced it in sway-git, I guess that implies this is a bigger issue with wl-roots itself. Should I close this issue, and make a issue over there?

**Comment by UltraBlackLinux** [comment]
> Just reproduced it in sway-git

same GPU driver...

>  Should I close this issue

yes

> make a issue

but where? Mesa maybe?

**Comment by nonetrix** [comment]
Well, I think this issue could be completely unrelated to wlroots, hyprland, wayland, or even mesa. I've recently discovered my power supply is most likely on it's death bed, luckily it didn't go up in flames or give the delightful magic smoke, but it appears to be dying. My computer just randomly turned off, seemingly for no reason very symptomatic of overheating, or power supply over voltage, or power supply failure in general etc. and this was following a bunch of random seemingly unexplainable issues. My computer already had the cold boot issue listed in the below Reddit post, but assumed it was a UEFI firmware bug, since it conveniently started happening right after I updated my UEFI ironically. However, I think my new GPU which consumes slightly more power than my old one was enough to push it over the edge, to the point where it started to have more issues than just a delayed boot.

https://www.reddit.com/r/Amd/comments/73vsj9/cold_boot_issues_it_might_be_your_psu/

Lessons learned, still could be something else unrelated my power supply. If it is unrelated, I'll attempt to investigate it further and report a issue where ever it might be relevant such was wlroots, wayland, mesa etc. Again, sorry I'm not great at making issues I'll admit.
